crowbar_framework/app/controllers/backups_controller.rb

```diff
@@ -0,0 +1,171 @@
+        format.html { redirect_to backups_path }
+      else
+        msg = I18n.t("backups.index.invalid_filename")
+        format.json { render json: { error: msg }, status: :bad_request }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Restore
+  #
+  # Provides the restful api call for
+  # /utils/backup/restore   POST   Trigger a restore
+  def restore
+    respond_to do |format|
+      if @backup.restore
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        format.json do
+          render json: { error: @backup.errors.full_messages.first }, status: :unprocessable_entity
+        end
+        format.html do
+          flash[:alert] = @backup.errors
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Download
+  #
+  # Provides the restful api call for
+  # /utils/backup/download/:name/:created_at 	GET 	Download a backup
+  def download
+    respond_to do |format|
+      if @backup.exist?
+        format.any do
+          send_file(
+            @backup.path,
+            filename: @backup.filename
+          )
+        end
+      else
+        msg = I18n.t("backups.index.missing_backup")
+        format.json { render json: { error: msg }, status: :not_found }
+        format.html do
+          flash[:alert] = msg
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Upload
+  #
+  # Provides the restful api call for
+  # /utils/backup/upload   POST   Upload a backup
+  def upload
+    @backup = Backup.new(backup_upload_params)
+
+    respond_to do |format|
+      if @backup.save
+        format.json { head :ok }
+        format.html { redirect_to backups_path }
+      else
+        format.json do
+          render json: { error: @backup.errors.full_messages.first }, status: :unprocessable_entity
+        end
+        format.html do
+          flash[:alert] = @backup.errors
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  #
+  # Delete Backups
+  #
+  # Provides the restful api call for
+  # data-confirm method delete
+  # /utils/backup/destroy 	DELETE 	Delete a backup
+  def destroy
+    respond_to do |format|
+      if @backup.destroy
+        format.json do
+          head :ok
+        end
+        format.html do
+          redirect_to backups_path
+        end
+      else
+        format.json do
+          render json: {
+            error: I18n.t("backups.destroy.failed")
+          }, status: :unprocessable_entity
+        end
+        format.html do
+          flash[:alert] = I18n.t("backups.destroy.failed")
+          redirect_to backups_path
+        end
+      end
+    end
+  end
+
+  protected
+
+  def set_backup
+    @backup = Backup.find(params[:id])
+  end
+
+  def backup_params
+    params.require(:backup).permit(:name)
+  end
+
+  def backup_upload_params
+    params.permit(:file)
```

Comment:
For backup params you are putting it into a namespace but not for the upload? Please be consistent.